esmvalcore/preprocessor/_io.py

```diff
@@ -90,6 +91,16 @@ def concatenate(cubes):
     concatenated = iris.cube.CubeList(cubes).concatenate()
     if len(concatenated) == 1:
         return concatenated[0]
+    if len(concatenated) == 2:
+        try:
+            concatenated[0].coord('time')
+            concatenated[1].coord('time')
+        except iris.exceptions.CoordinateNotFoundError:
+            pass
+        else:
+            concatenated = _concatenate_two_overlapping_cubes(concatenated)
+            if len(concatenated) == 1:
```

Comment:
these two lines are duplicated, if you move the `if len(concatenated) == 2` block before the `if len(concatenated) == 1` statement above, they are not needed.

Reply:
no, there are two possible cases stemming from two different concatenation processes: the first if check - `if len(concatenated) == 1: return concatenated[0]` comes from simple concatenation and if that returns false then we check on the length to be exactly 2 - `if len(concatenated) == 2: try: except: else: if len(concatenated) == 1: return concatenated[0]` note that the seemingly repetive check on the lenght = 1 comes from the `_concatenate_two_overlapping_cubes`

Reply:
There is really no need to check `if len(concatenated) == 1` two times, can you have another look please?